File: apps/server/Generators/AliasVault.Generators.Identity/Implementations/IdentityGeneratorEn.cs
//-----------------------------------------------------------------------
// <copyright file="IdentityGeneratorEn.cs" company="lanedirt">
// Copyright (c) lanedirt. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
// </copyright>
//-----------------------------------------------------------------------

namespace AliasVault.Generators.Identity.Implementations;

using AliasVault.Generators.Identity.Implementations.Base;

/// <summary>
/// Dutch identity generator which implements IIdentityGenerator and generates
/// random dutch identities.
/// </summary>
public class IdentityGeneratorEn : IdentityGenerator
{
    /// <inheritdoc cref="IdentityGenerator.FirstNamesListMale" />
    protected override string FirstNamesListMale => "AliasVault.Generators.Identity.Implementations.Dictionaries.en.firstnames_male";

    /// <inheritdoc cref="IdentityGenerator.FirstNamesListFemale" />
    protected override string FirstNamesListFemale => "AliasVault.Generators.Identity.Implementations.Dictionaries.en.firstnames_female";

    /// <inheritdoc cref="IdentityGenerator.LastNamesList" />
    protected override string LastNamesList => "AliasVault.Generators.Identity.Implementations.Dictionaries.en.lastnames";
}

File: shared/identity-generator/src/implementations/IdentityGeneratorEn.ts
import { IdentityGenerator } from "./base/IdentityGenerator";
import maleNames from '../dictionaries/en/firstnames_male';
import femaleNames from '../dictionaries/en/firstnames_female';
import lastNames from '../dictionaries/en/lastnames';

/**
 * Identity generator for English language using English word dictionaries.
 */
export class IdentityGeneratorEn extends IdentityGenerator {
  /**
   * Get the male first names.
   */
  protected getFirstNamesMaleJson(): string[] {
    return maleNames;
  }

  /**
   * Get the female first names.
   */
  protected getFirstNamesFemaleJson(): string[] {
    return femaleNames;
  }

  /**
   * Get the last names.
   */
  protected getLastNamesJson(): string[] {
    return lastNames;
  }
}
